lite2/client.go

```diff
@@ -1004,7 +1031,7 @@ func (c *Client) signedHeaderFromPrimary(height int64) (*types.SignedHeader, err
 		if err == nil {
 			// sanity check
 			if height > 0 && h.Height != height {
-				return nil, errors.Errorf("expected %d height, got %d", height, h.Height)
+				return nil, fmt.Errorf("expected %d height, got %d: %w", height, h.Height, err)
```

Comment:
err equals nil here, no? why log it then?

Reply:
you're right - I can remove it